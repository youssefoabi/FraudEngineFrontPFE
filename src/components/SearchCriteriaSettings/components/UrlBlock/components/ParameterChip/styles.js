import { eucalyptus, trinidad, white } from '@cdiscount/ui-colors';

export default function style() {
  return {
    statusLinkedChip: {
      backgroundColor: eucalyptus,
      borderRadius: 5,
      color: white,
      marginLeft: 5,
      marginTop: 10,
      height: 25,
    },
    statusNotLinkedChip: {
      backgroundColor: trinidad,
      borderRadius: 5,
      color: white,
      marginLeft: 5,
      marginTop: 10,
      height: 25,
    },
  };
}
