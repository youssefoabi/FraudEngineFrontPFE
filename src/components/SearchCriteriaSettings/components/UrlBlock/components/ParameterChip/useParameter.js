import { findIndex, flatten, map, pipe, prop, values } from 'ramda';
import { isNilOrEmpty } from 'ramda-adjunct';
import { useMemo } from 'react';

const emptyObject = {};

const useParameter = ({ scopeId, engine, parameter }) => {
  const { inputFields } = engine || emptyObject;

  const isLinkedParameter = useMemo(() => {
    if (isNilOrEmpty(inputFields)) return false;

    const engineParameters = pipe(values, map(prop('parameters')), flatten)(inputFields);

    if (isNilOrEmpty(engineParameters)) return false;

    const foundParameter =
      findIndex(
        enginePram => parameter === prop('key', enginePram) && scopeId === prop('scopeId', enginePram),
      )(engineParameters) !== -1;

    return foundParameter;
  }, [inputFields, scopeId, parameter]);

  return {
    isLinkedParameter,
  };
};

export default useParameter;
