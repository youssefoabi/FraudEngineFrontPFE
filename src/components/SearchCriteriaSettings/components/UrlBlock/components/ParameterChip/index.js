import { number, shape, string } from 'prop-types';
import { useStyles } from '@cdiscount/backoffice-ui';
import MuiChip from '@material-ui/core/Chip';
import React from 'react';

import styles from './styles';
import useParameter from './useParameter';

const ParameterChip = ({ scopeId, parameter, engine }) => {
  const { isLinkedParameter } = useParameter({
    engine,
    parameter,
    scopeId,
  });

  const classes = useStyles(styles);

  return (
    <MuiChip
      label={parameter}
      className={isLinkedParameter ? classes.statusLinkedChip : classes.statusNotLinkedChip}
    />
  );
};

ParameterChip.propTypes = {
  engine: shape({}),
  parameter: string,
  scopeId: number,
};

ParameterChip.defaultProps = {
  engine: null,
  parameter: null,
  scopeId: null,
};

export default ParameterChip;
