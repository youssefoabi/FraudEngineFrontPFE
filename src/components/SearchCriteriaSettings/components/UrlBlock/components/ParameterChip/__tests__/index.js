import { itRendersAllMutations } from '@cdiscount/jest-utils';

import ParameterChip from '../';

jest.resetModules();

jest.mock('../useParameter', () => ({
  __esModule: true,
  default: jest
    .fn()
    .mockReturnValueOnce({})
    .mockReturnValueOnce({
      isLinkedParameter: true,
    })
    .mockReturnValueOnce({
      isLinkedParameter: false,
    }),
  namedExport: 'useParameter',
}));

const mutations = [
  { name: 'without props' },
  { name: 'with linked parameter', props: { parameter: 'Search' } },
  { name: 'with unlinked parameter', props: { parameter: 'Search' } },
];

describe(ParameterChip.name, () => {
  itRendersAllMutations(ParameterChip, mutations);
});
