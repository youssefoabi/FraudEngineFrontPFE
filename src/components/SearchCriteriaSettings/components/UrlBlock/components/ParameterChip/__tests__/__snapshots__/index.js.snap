// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParameterChip ParameterChip renders correctly with linked parameter 1`] = `
<WithStyles(ForwardRef(Chip))
  className="makeStyles-statusLinkedChip-3"
  label="Search"
/>
`;

exports[`ParameterChip ParameterChip renders correctly with unlinked parameter 1`] = `
<WithStyles(ForwardRef(Chip))
  className="makeStyles-statusNotLinkedChip-6"
  label="Search"
/>
`;

exports[`ParameterChip ParameterChip renders correctly without props 1`] = `
<WithStyles(ForwardRef(Chip))
  className="makeStyles-statusNotLinkedChip-2"
  label={null}
/>
`;
